android/src-huawei/com/badlogic/gdx/pay/app/AndroidLauncher.java

```diff
@@ -0,0 +1,118 @@
+package com.badlogic.gdx.pay.app;
+
+import android.content.Intent;
+import android.content.IntentSender;
+import android.util.Log;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.pay.Transaction;
+import com.badlogic.gdx.pay.android.huawei.HuaweiPurchaseManager;
+import com.badlogic.gdx.pay.android.huawei.IAPListener;
+import com.huawei.hms.iap.Iap;
+import com.huawei.hms.iap.IapApiException;
+import com.huawei.hms.iap.IapClient;
+import com.huawei.hms.iap.entity.ConsumeOwnedPurchaseResult;
+import com.huawei.hms.iap.entity.InAppPurchaseData;
+import com.huawei.hms.iap.entity.OrderStatusCode;
+import com.huawei.hms.iap.entity.PurchaseIntentResult;
+import com.huawei.hms.iap.entity.PurchaseResultInfo;
+
+import org.json.JSONException;
+
+/**
+ * Created by Francesco Stranieri on 11.05.2020.
+ */
+
+public class AndroidLauncher extends GenericAndroidLauncher implements IAPListener {
+
+    private final int PURCHASE_STATUS_RESULT_CODE = 7265;
+
+    GdxPayApp game;
+
+    @Override
+    protected void initFlavor(GdxPayApp game) {
+        this.game = game;
+        super.initFlavor(this.game);
+
+        this.game.purchaseManager = new HuaweiPurchaseManager(this, this);
+    }
+
+    @Override
+    public void onRegionNotSupported() {
+        Gdx.app.log("HUAWEI", "REGION NOT SUPPORTED");
+    }
+
+    @Override
+    public void onLoginRequired() {
+        Gdx.app.log("HUAWEI", "LOGIN REQUESTED");
+    }
+
+    @Override
+    public void onIAPError(IapApiException exception) {
+        Gdx.app.error("HUAWEI", "IAP EXC", exception);
+    }
+
+    @Override
+    public void onError(Exception exception) {
+        Gdx.app.error("HUAWEI", "IAP ERR", exception);
+    }
+
+    @Override
+    public void onPurchaseResult(PurchaseIntentResult result) {
+        try {
+            result.getStatus().startResolutionForResult(this, this.PURCHASE_STATUS_RESULT_CODE);
+        } catch (IntentSender.SendIntentException e) {
+        }
+    }
+
+    @Override
+    public void onConsumedResult(ConsumeOwnedPurchaseResult result) {
+
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
use `AndroidApplication.addAndroidEventListener` method to install a hook to this callback from HuaweiPurchaseManager.